rdr_service/tools/tool_libs/cope_answers.py

```diff
@@ -0,0 +1,134 @@
+#! /bin/env python
+#
+# Find COPE survey response counts and breakdowns for each question
+#
+
+import argparse
+import json
+import logging
+import random
+import sys
+
+from rdr_service.dao import database_factory
+from rdr_service.services.gcp_utils import gcp_format_sql_instance, gcp_make_auth_header
+from rdr_service.services.system_utils import make_api_request, setup_logging, setup_i18n
+from rdr_service.tools.tool_libs import GCPProcessContext
+
+_logger = logging.getLogger("rdr_logger")
+
+# Tool_cmd and tool_desc name are required.
+# Remember to add/update bash completion in 'tool_lib/tools.bash'
+tool_cmd = "cope-answers"
+tool_desc = "create a json text file that gives total answer counts for "
+
+ANSWER_COUNT_SQL = """
+    SELECT qc.value question_code, ac.value answer_code, COUNT(DISTINCT qr.participant_id) answer_count
+    FROM questionnaire_response_answer qra
+    INNER JOIN questionnaire_response qr ON qr.questionnaire_response_id = qra.questionnaire_response_id
+    INNER JOIN questionnaire_question qq ON qq.questionnaire_question_id = qra.question_id
+    INNER JOIN code qc ON qc.code_id = qq.code_id
+    INNER JOIN code ac ON ac.code_id = qra.value_code_id
+    INNER JOIN questionnaire_history qh ON qh.questionnaire_id = qr.questionnaire_id
+        AND qh.version = qr.questionnaire_version
+    WHERE qc.value IN ('msds_11', 'msds_12', 'copect_40_xx15', 'cdc_covid_19_18', 'phq_9_4',
+                      'ipaq_1', 'ipaq_3', 'ipaq_5', 'ipaq_7', 'ucla_ls8_9')
+        AND qra.end_time IS NULL
+        AND qh.last_modified > :cope_month
+    GROUP BY qc.value, ac.value;
+"""
+
+
+class CopeAnswersClass(object):
+    def __init__(self, args, gcp_env):
+        self.args = args
+        self.gcp_env = gcp_env
+
+    def _setup_database_connection(self):
+        _logger.info("retrieving db configuration...")
+        headers = gcp_make_auth_header()
+        resp_code, resp_data = make_api_request(
+            "{0}.appspot.com".format(self.gcp_env.project), "/rdr/v1/Config/db_config", headers=headers
+        )
+        if resp_code != 200:
+            _logger.error(resp_data)
+            _logger.error("failed to retrieve config, aborting.")
+            return 1
+
+        passwd = resp_data["rdr_db_password"]
+        if not passwd:
+            _logger.error("failed to retrieve database user password from config.")
+            return 1
```

Comment:
This can be simplified to `passwd = self.gcp_env.get_app_db_config()['rdr_db_password']`